src/ucp/core/ucp_ep.c

```diff
@@ -1338,30 +1346,46 @@ ucs_status_t ucp_ep_config_init(ucp_worker_h worker, ucp_ep_config_t *config,
     }
 
     /* configuration for rndv */
-    rndv_max_bw = 0;
+    get_zcopy_lane_count = 0;
+    put_zcopy_lane_count = 0;
+    rndv_max_bw          = 0;
     for (i = 0; (i < config->key.num_lanes) &&
                 (config->key.rma_bw_lanes[i] != UCP_NULL_LANE); ++i) {
         lane      = config->key.rma_bw_lanes[i];
         rsc_index = config->key.lanes[lane].rsc_index;
 
         if (rsc_index != UCP_NULL_RESOURCE) {
             iface_attr = ucp_worker_iface_get_attr(worker, rsc_index);
+            md_attr    = &context->tl_mds[context->tl_rscs[rsc_index].md_index].attr;
+
+            /* Rkey_ptr */
+            if (md_attr->cap.flags & UCT_MD_FLAG_RKEY_PTR) {
+                config->tag.rndv.rkey_ptr_dst_mds |=
+                        UCS_BIT(config->key.lanes[lane].dst_md_index);
+            }
 
             /* GET Zcopy */
-            config->tag.rndv.min_get_zcopy = ucs_max(config->tag.rndv.min_get_zcopy,
-                                                     iface_attr->cap.get.min_zcopy);
+            if (iface_attr->cap.flags & UCT_IFACE_FLAG_GET_ZCOPY) {
+                config->tag.rndv.min_get_zcopy = ucs_max(config->tag.rndv.min_get_zcopy,
+                                                         iface_attr->cap.get.min_zcopy);
 
-            config->tag.rndv.max_get_zcopy = ucs_min(config->tag.rndv.max_get_zcopy,
-                                                     iface_attr->cap.get.max_zcopy);
+                config->tag.rndv.max_get_zcopy = ucs_min(config->tag.rndv.max_get_zcopy,
+                                                         iface_attr->cap.get.max_zcopy);
+                config->tag.rndv.get_zcopy_lanes[get_zcopy_lane_count++] = lane;
```

Comment:
let's add:
```
ucs_assert(get_zcopy_lane_count <= UCP_MAX_LANES);

```

Reply:
fixed (but it should be < rather than <= since it's pre-increment)